iron-form.html

```diff
@@ -11,503 +11,459 @@
-     *
-     * @param {Event} event A `reset` event.
-     */
-    _onReset: function(event) {
-      this._resetCustomElements();
-    },
-
-    /**
-     * Returns a json object containing name/value pairs for all the registered
-     * custom components and native elements of the form. If there are elements
-     * with duplicate names, then their values will get aggregated into an
-     * array of values.
-     *
-     * @return {!Object}
-     */
-    serialize: function() {
-      var json = {};
-
-      function addSerializedElement(name, value) {
-        // If the name doesn't exist, add it. Otherwise, serialize it to
-        // an array,
-        if (json[name] === undefined) {
-          json[name] = value;
-        } else {
-          if (!Array.isArray(json[name])) {
-            json[name] = [json[name]];
+        // Start by making the form check the native elements it knows about.
+        var valid = this._form.checkValidity();
+        var elements = this._getValidatableElements();
+
+        // Go through all the elements, and validate the custom ones.
+        for (var el, i = 0; el = elements[i], i < elements.length; i++) {
+          // This is weird to appease the compiler. We assume the custom element
+          // has a validate() method, otherwise we can't check it.
+          var validatable = /** @type {{validate: (function() : boolean)}} */ (el);
+          if (validatable.validate) {
+            valid = !!validatable.validate() && valid;
           }
-          json[name].push(value);
         }
-      }
+        return valid;
+      },
 
-      // Go through all of the registered custom components.
-      for (var el, i = 0; el = this._customElements[i], i < this._customElements.length; i++) {
-        // If this custom element is inside a custom element that has already
-        // registered to this form, skip it.
-        if (!this._isChildOfRegisteredParent(el, true) && this._useValue(el)) {
-          addSerializedElement(el.name, el.value);
+      /**
+       * Submits the form.
+       */
+      submit: function(event) {
+        // We are not using this form for submission, so always cancel its event.
+        if (event)
+          event.preventDefault();
+
+        // If you've called this before distribution happened, bail out.
+        if (!this._form || !this.validate()) {
+          return;
         }
-      }
 
-      // Also go through the form's native elements.
-      for (var el, i = 0; el = this.elements[i], i < this.elements.length; i++) {
-        // If this native element is inside a custom element that has already
-        // registered to this form, skip it.
-        if (this._isChildOfRegisteredParent(el, true) || !this._useValue(el)) {
-          continue;
+        // Remove any existing children in the submission form (from a previous submit).
+        while (this.$.helper.children.length !== 0) {
+          this.$.helper.removeChild(this.$.helper.children[0]);
         }
```

Comment:
How about doing `this.$.helper.textContent = ''` to remove all the content inside `helper`?